compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs

```diff
@@ -393,6 +398,83 @@ fn trait_constraint_eq(
     Ok(Value::Bool(constraint_a == constraint_b))
 }
 
+// fn zeroed<T>() -> T
+fn zeroed(return_type: Type, location: Location) -> IResult<Value> {
+    match return_type {
+        Type::FieldElement => Ok(Value::Field(0u128.into())),
+        Type::Array(length_type, elem) => {
+            if let Some(length) = length_type.evaluate_to_u32() {
+                let element = zeroed(elem.as_ref().clone(), location)?;
+                let array = std::iter::repeat(element).take(length as usize).collect();
+                Ok(Value::Array(array, Type::Array(length_type, elem)))
+            } else {
+                Err(InterpreterError::NonIntegerArrayLength { typ: *length_type, location })
+            }
+        }
+        Type::Slice(_) => Ok(Value::Slice(im::Vector::new(), return_type)),
+        Type::Integer(sign, bits) => match (sign, bits) {
+            (Signedness::Unsigned, IntegerBitSize::One) => Ok(Value::U8(0)),
+            (Signedness::Unsigned, IntegerBitSize::Eight) => Ok(Value::U8(0)),
+            (Signedness::Unsigned, IntegerBitSize::Sixteen) => Ok(Value::U16(0)),
+            (Signedness::Unsigned, IntegerBitSize::ThirtyTwo) => Ok(Value::U32(0)),
+            (Signedness::Unsigned, IntegerBitSize::SixtyFour) => Ok(Value::U64(0)),
+            (Signedness::Signed, IntegerBitSize::One) => Ok(Value::I8(0)),
+            (Signedness::Signed, IntegerBitSize::Eight) => Ok(Value::I8(0)),
+            (Signedness::Signed, IntegerBitSize::Sixteen) => Ok(Value::I16(0)),
+            (Signedness::Signed, IntegerBitSize::ThirtyTwo) => Ok(Value::I32(0)),
+            (Signedness::Signed, IntegerBitSize::SixtyFour) => Ok(Value::I64(0)),
+        },
+        Type::Bool => Ok(Value::Bool(false)),
+        Type::String(length_type) => {
+            if let Some(length) = length_type.evaluate_to_u32() {
+                Ok(Value::String(Rc::new("\0".repeat(length as usize))))
+            } else {
+                Err(InterpreterError::NonIntegerArrayLength { typ: *length_type, location })
+            }
+        }
+        Type::FmtString(_, _) => {
+            let item = "format strings in a comptime context".into();
+            Err(InterpreterError::Unimplemented { item, location })
```

Comment:
Why not return `Value::Zeroed` here and in the cases where the array/string length is unknown?

Reply:
For format strings we should return an actual format string value instead I think.
I want to limit the scope of `Value::Zeroed` as much as possible since I think it is a hack.
I made the change to the array case though.